modules/images/dominant-color/load.php

```diff
@@ -0,0 +1,376 @@
+<?php
+/**
+ * Module Name: Dominant Color
+ * Description: Adds support to store dominant-color for an image and create a placeholder background with that color.
+ * Experimental: Yes
+ *
+ * @package performance-lab
+ * @since n.e.x.t
+ */
+
+/**
+ * Add the dominant color metadata to the attachment.
+ *
+ * @since n.e.x.t
+ *
+ * @param array $metadata The attachment metadata.
+ * @param int   $attachment_id The attachment ID.
+ *
+ * @return array $metadata
+ */
+function dominant_color_metadata( $metadata, $attachment_id ) {
+	if ( ! wp_attachment_is_image( $attachment_id ) ) {
+
+		return $metadata;
+	}
+
+	$dominant_color = dominant_color_get( $attachment_id );
+
+	if ( ! empty( $dominant_color ) ) {
+		$metadata['dominant_color'] = $dominant_color;
+	}
+
+	$has_transparency = dominant_color_get_has_transparency( $attachment_id );
+
+	if ( ! empty( $has_transparency ) ) {
+		$metadata['has_transparency'] = $has_transparency;
+	} else {
+		$metadata['has_transparency'] = false;
+	}
+
+	return $metadata;
+}
+add_filter( 'wp_generate_attachment_metadata', 'dominant_color_metadata', 10, 2 );
+
+/**
+ * Filter various image attributes to add the dominant color to the image
+ *
+ * @since n.e.x.t
+ *
+ * @param array  $attr Attributes for the image markup.
+ * @param object $attachment Image attachment post.
+ *
+ * @return mixed
+ */
+function dominant_color_tag_add_adjust_to_image_attributes( $attr, $attachment ) {
```

Comment:
Let's name this function something more descriptive that would also work in core, i.e. with a simple `wp_` prefix: I suggest `dominant_color_update_attachment_image_attributes`.